### DISCUSSION TITLE
Does every service need an SSP?

### DISCUSSION THREAD

**Discussion by amirshadaab** [oscal-club/community#37] [Q&A]
Forgive my ignorance on the topic. I am new to compliance world. I am a traditional software developer who is trying to model our Federal compliance needs. What I am reading makes sense. Here is a high level overview of my understanding of the Implementation Layer:
- A catalog is a collection of controls
- A profile is a subset of those controls
- New profiles can be created by importing from other profiles and catalogs
- SSP is defining how these controls are met via components which can be composed of other components

So I have some questions:
- What is the difference between SSP and Component Definition? Do I need both?
- Does SSP describe the whole entire system? Can this be done at a service level? Meaning catalogs, profiles, controls, components, etc. can be defined at a global level and each service has a different path to become compliant for a specific profile. If so, do I need to have SSP per service? Our company is made up of 100s if not 1000s of services. Writing an SSP covering everything is going to be a nightmare. I am missing the piece of how these concepts are applied to different services.

**Comment by pjavan** [comment]
SSPs contains details for how controls are implemented

The Component Definition (CD) offer controls in two states:

- CAN DO - Component definition describes the capabilities of the component without regard for any specific framework. This is the world of possibility.  (e.g., Enumerate all the cryptographic modules available.)

- SHOULD DO - Identifies a specific framework or target requirement set:
    -     Provide the configuration instructions and/or scripts for deploying the component in a manner that satisfies the framework/requirement set.
    -     Provide the write-up for each control satisfied assuming the provided instructions/scripts are used
